frontend/src/components/institute/NewSchoolPage.vue

```diff
@@ -0,0 +1,519 @@
+        physicalAddrCity: null,
+        physicalAddrProvince: null,
+        physicalAddrCountry: null,
+        physicalAddrPostal: null,
+      },
+      rules: Rules,
+      newSchoolOpenDatePicker: null,
+      schoolFacilityTypes: [],
+      schoolCategoryTypes: [],
+      schoolOrganizationTypes: [],
+      schoolNeighborhoodLearningTypes: [],
+      schoolGradeTypes: [],
+      sameAsMailingCheckbox: true,
+      provinceCodeValues: [],
+      countryCodeValues: [],
+
+    };
+  },
+  computed: {
+    ...mapGetters('auth', ['isAuthenticated','userInfo']),
+    ...mapState('institute', ['facilityTypeCodes']),
+    ...mapState('institute', ['schoolCategoryTypeCodes']),
+    ...mapState('institute', ['schoolOrganizationTypeCodes']),
+    ...mapState('institute', ['schoolNeighborhoodLearningCodes']),
+    ...mapState('institute', ['gradeCodes']),
+    ...mapState('institute', ['provinceCodes']),
+    ...mapState('institute', ['countryCodes']),
+    ...mapState('institute', ['schoolCategoryFacilityTypesMap']),
+
+    allowedFacilityTypeCodesForSchoolCategoryCode(){
+      if(this.newSchool?.categoryCode){
+        return this.schoolCategoryFacilityTypesMap[this.newSchool?.categoryCode]?.map(schoolCatFacilityTypeCode =>  this.facilityTypeCodes.find(facTypCode=> facTypCode.facilityTypeCode === schoolCatFacilityTypeCode));
+      }else {
+        return [];
+      }
+    },
+  },
+  created() {
+    this.$store.dispatch('institute/getFacilityTypeCodes').then(() => {
+      this.schoolFacilityTypes = this.facilityTypeCodes;
+    });
+    this.$store.dispatch('institute/getSchoolCategoryTypeCodes').then(() => {
+      this.schoolCategoryTypes = this.schoolCategoryTypeCodes;
+    });
+    this.$store.dispatch('institute/getSchoolOrganizationTypeCodes').then(() => {
+      this.schoolOrganizationTypes = this.schoolOrganizationTypeCodes;
+    });
+    this.$store.dispatch('institute/getSchoolNeighborhoodLearningCodes').then(() => {
+      this.schoolNeighborhoodLearningTypes = this.schoolNeighborhoodLearningCodes;
+    });
+    this.$store.dispatch('institute/getGradeCodes').then(() => {
+      this.schoolGradeTypes = this.gradeCodes;
+    });
+    this.$store.dispatch('institute/getProvinceCodes').then(() => {
+      this.provinceCodeValues = this.provinceCodes.filter(province =>  province.provinceCode === 'BC' || province.provinceCode === 'YT');
+    });
+    this.$store.dispatch('institute/getCountryCodes').then(() => {
+      this.countryCodeValues = this.countryCodes;
+    });
+    this.$store.dispatch('institute/getSchoolCategoryFacilityTypesMap');
+  },
+  methods: {
+    saveNewSchoolOpenDate(date) {
+      this.$refs.newSchoolOpenDateFilter.save(date);//TODO
+    },
+    closeNewSchoolPage() {
+      this.resetForm();
+      this.$emit('newSchool:closeNewSchoolPage');
+    },
+    addNewSchool() {
+      this.processing = true;
+
+      ApiService.apiAxios.post(`${Routes.institute.ROOT_ENDPOINT}/????`, this.newSchool)
```

Comment:
URL looks incorrect... is the /???? a placeholder for a future end-point?

Reply:
This ticket was just for the front end UI work. I left the url like that so its obvious for whoever does the backend